src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/SqlJournal.cs

```diff
@@ -282,19 +282,26 @@ protected bool WaitingForInitialization(object message)
                 .WasHandled;
         }
 
-        private async Task<AllPersistenceIds> Initialize()
+        private async Task<object> Initialize()
         {
-            using (var connection = CreateDbConnection())
+            try
             {
-                await connection.OpenAsync();
-
-                if (_settings.AutoInitialize)
+                using (var connection = CreateDbConnection())
                 {
-                    await QueryExecutor.CreateTablesAsync(connection, _pendingRequestsCancellation.Token);
-                }
+                    await connection.OpenAsync();
 
-                var ids = await QueryExecutor.SelectAllPersistenceIdsAsync(connection, _pendingRequestsCancellation.Token);
-                return new AllPersistenceIds(ids);
+                    if (_settings.AutoInitialize)
+                    {
+                        await QueryExecutor.CreateTablesAsync(connection, _pendingRequestsCancellation.Token);
+                    }
+
+                    var ids = await QueryExecutor.SelectAllPersistenceIdsAsync(connection, _pendingRequestsCancellation.Token);
+                    return new AllPersistenceIds(ids);
+                }
+            }
+            catch (Exception e)
+            {
+                return new Failure {Exception = e};
```

Comment:
The result of the `Initialize()` method is passed via `PipeTo()` to `WaitForInitialization(object message)`.  Previous behavior was that it would only pass the result `AllPersistenceIds(ids)`, regardless of the success or failure of the previous calls.  This change adds handling of any possible exceptions and population of a `Failure` message that is passed to `WaitForInitialization`,  which handles the exception and logs it appropriately.